eva_submission/eload_validation.py

```diff
@@ -432,64 +440,35 @@ def _collect_normalisation_check_results(self, vcf_files, output_dir):
-                validator_output_file = os.path.join(self._get_dir('vcf_check'), vcf_name + '.vcf_validator.txt')
-                has_sv_per_vcf = self._detect_structural_variant_from_validator_output(validator_output_file)
+            vcf_name = os.path.basename(vcf_file)
+
+            tmp_sv_check_log = resolve_single_file_path(
+                os.path.join(output_dir, 'sv_check',  vcf_name + '.sv_check.log')
+            )
+            tmp_sv_check_sv_vcf = resolve_single_file_path(
+                os.path.join(output_dir, 'sv_check', vcf_name + '.sv_list.vcf')
+            )
+
+            # move the output files
+            sv_check_log = self._move_file(
+                tmp_sv_check_log,
+                os.path.join(self._get_dir('sv_check'), vcf_name + '.sv_check.log')
+            )
+            sv_check_sv_vcf = self._move_file(
+                tmp_sv_check_sv_vcf,
+                os.path.join(self._get_dir('sv_check'), vcf_name + '.sv_list.vcf')
+            )
+
+            if sv_check_log and sv_check_sv_vcf:
+                nb_sv = self.parse_sv_check_log(sv_check_log)
             self.eload_cfg.set('validation', 'structural_variant_check', 'files', os.path.basename(vcf_file),
-                               value={'has_structural_variant': has_sv_per_vcf})
+                               value={'has_structural_variant': nb_sv > 0, 'number_sv': nb_sv})
         self.eload_cfg.set('validation', 'structural_variant_check', 'pass', value=True)
 
-    def _detect_structural_variant_from_variant_lines(self, vcf_file):
-        no_of_variant_lines_per_vcf = 0
-        has_sv_per_vcf = False
-        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Pages: 6, 16)
-        symbolic_allele_pattern = "^<(DEL|INS|DUP|INV|CNV|BND)"
-        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 17)
-        complex_rearrangements_breakend_pattern = "^[ATCGNatgcn]+\[.+:.+\[$|^[ATCGNatgcn]+\].+:.+\]$|^\].+:.+\][ATCGNatgcn]+$|^\[.+:.+\[[ATCGNatgcn]+$"
-        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 18)
-        complex_rearrangements_special_breakend_pattern = "^[ATGCNatgcn]+<[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*>$"
-        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Page: 22)
-        single_breakend_pattern = "^\.[ATGCNatgcn]+|[ATGCNatgcn]+\.$"
-        sv_regex = re.compile(f'{symbolic_allele_pattern}|{complex_rearrangements_breakend_pattern}|'
-                              f'{complex_rearrangements_special_breakend_pattern}|{single_breakend_pattern}')
-        if vcf_file.endswith('.vcf.gz'):
-            open_file = gzip.open(vcf_file, mode="rt")
-        else:
-            open_file = open(vcf_file, mode="r")
-        for file_line in open_file:
-            if file_line[0] == "#":
-                continue;
-            no_of_variant_lines_per_vcf = no_of_variant_lines_per_vcf + 1
-            if no_of_variant_lines_per_vcf > 10000:
```

Comment:
I'm guessing we've removed this limit because the process is running in nextflow now, in parallel with the other validation tasks... is that right?